### ISSUE TITLE
optimize travis prcess

### ISSUE THREAD

**Issue by romani** [sevntu-checkstyle/sevntu.checkstyle#519]
1) 
review https://github.com/checkstyle/sonar-checkstyle/blob/master/.travis.yml , split of sevntu travis stuff into two files will give user ability to launch easily build item on his side.

2)
we need to question why we do "javadoc:javadoc" as separate step.
"mvn clean install" should be be enough.

**Comment by rnveach** [comment]
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/b3f59ada0f26b59d3c33cdd61db5a9bc163ab178/.travis.yml#L22-L23
>         - CMD0="git clone git://git.code.sf.net/p/eclipse-cs/git eclipse-cs-git"
>         - CMD1="&& cd eclipse-cs-git/ && git checkout 7.2.0"

This can also be optimized to cache eclipse-cs and only do fetch/reset. It must take care to notice that if clone doesn't exist, it must re-clone it itself.

**Comment by romani** [comment]
We never do caching of clones , as clones are quick in cloud. Extra logic in CI configuration is not good for human. The simpler the better.

**Comment by rnveach** [comment]
> 1.
> split of sevntu travis stuff into two files will give user ability to launch easily build item on his side

I feel this is done. We have a validation.sh file and a no-exception one.
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/.ci/validation.sh
https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/.ci/checks-no-exception.sh

> 2.
> we need to question why we do "javadoc:javadoc" as separate step.
"mvn clean install" should be be enough.

I only did my tests on `sevntu-checks`.

Looking at them both, install does run javadoc. On my local I did notice a slight discrepancy between the 2.

Install:
`Generating M:\checkstyleWorkspaceEclipse\sevntu\sevntu-checks\target\apidocs\com\github\sevntu\checkstyle\SevntuUtil.html...`
Javadoc:
`Generating M:\checkstyleWorkspaceEclipse\sevntu\sevntu-checks\target\site\apidocs\com\github\sevntu\checkstyle\SevntuUtil.html...`

It is hard to tell, but Javadoc is placing `apidocs` into a `site` folder while install puts `apidocs` in the root of target.

I did confirm `mvn verify` is doing almost identically the same thing as `mvn install` (minus the install process).

Should we swap everything to install, or is verify and dropping the others enough, or ...?

**Comment by rnveach** [comment]
Javadoc:Javadoc was first added in 8f4ac566efe1e697207e8f202993b715cb5ed109 which links back to https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/409 .

It seems it was added to avoid future problems. Look at the pom at the time ( https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/8f4ac566efe1e697207e8f202993b715cb5ed109/sevntu-checks/pom.xml ) I am not seeing a javadoc plugin. So maybe we couldn't run it via install/verify until we did.
